akka-actor/src/main/scala/akka/actor/ActorRef.scala

```diff
@@ -476,6 +476,47 @@ private[akka] trait MinimalActorRef extends InternalActorRef with LocalRef {
   protected def writeReplace(): AnyRef = SerializedActorRef(this)
 }
 
+/**
+ * An ActorRef that ignores any incoming messages.
+ *
+ * INTERNAL API
+ */
+private[akka] class IgnoreActorRef(systemName: String) extends MinimalActorRef {
+
+  override def provider: ActorRefProvider =
+    throw new UnsupportedOperationException("no provider")
+
+  override def path: ActorPath = IgnoreActorRef.actorPath(systemName)
+
+}
+
+/**
+ * INTERNAL API
+ */
+private[akka] object IgnoreActorRef {
+
+  /**
+   * Returns the ignore actor path for the passed actor system name.
+   * @param systemName
+   * @return
+   */
+  def actorPath(systemName: String): ActorPath =
+    RootActorPath(Address("akka", systemName)) / "system" / "ignore"
+
+  /**
+   * Check if the passed `path` has the same relative path of the IgnoreActorRef (ie: /system/ignore).
+   * @param path
+   * @return
+   */
+  def isIgnoreRefPath(path: String): Boolean = {
+    val pathElements = ActorPath.fromString(path).toStringWithoutAddress
+    // whatever sysName is good, we ony want to take the relative path
+    val ignorePathElements = actorPath("any-system").toStringWithoutAddress
+    pathElements == ignorePathElements
```

Comment:
To make this more efficient we don't have to make it an ActorPath but can use some string matching.